### PR TITLE
Rich text composer

### PR DESCRIPTION
**Pull request by ManhNTX**
https://user-images.githubusercontent.com/107173849/191901801-4b8698e2-cfe0-4648-aeed-f1a1815af0c3.mov

### CODE REVIEW

**Comment by dab246** [comment]
It would be great if you could provide a demo video of it.

**Comment by dab246** [comment]
Is it possible to set the background color of a rich text button to be the same as the color of the keyboard?

Screen Shot 2022-09-23 at 13 30 41

**Comment by ManhNTX** [comment]
> Is it possible to set the background color of a rich text button to be the same as the color of the keyboard?
> 
> Screen Shot 2022-09-23 at 13 30 41

Background toolbar setting by `backgroundKeyboardToolBarColor ` property.  I will try to get the background keyboard and update it in the next PR

**Review by hoangdat — COMMENTED** [review]
please remove merge commit, we should rebase

File: README.md [hunk 1]
```diff
@@ -1 +1,72 @@
...
+this README's contents appear on the landing page for your package.
+
+For information about how to write a good package README, see the guide for
+[writing package pages](https://dart.dev/guides/libraries/writing-package-pages). 
+
+For general information about developing packages, see the Dart guide for
+[creating packages](https://dart.dev/guides/libraries/create-library-packages)
+and the Flutter guide for
+[developing packages and plugins](https://flutter.dev/developing-packages). 
+-->
```
Comment (hoangdat) [inline README.md:12]: please remove it

File: README.md [hunk 2]
```diff
@@ -1 +1,72 @@
...
+    git:
+      url: https://github.com/linagora/rich-text-composer.git
+      ref: master
+ ```
+You should then run flutter packages upgrade or update your packages in IntelliJ.
+
+## Usage
+
+TODO: Include short and useful examples for package users. Add longer examples
+to `/example` folder.
```
Comment (hoangdat) [inline README.md:40]: what is TODO?

File: example/.metadata [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# This file tracks properties of this Flutter project.
```
Comment (hoangdat) [inline example/.metadata:1]: remove

File: example/README.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# example
```
Comment (hoangdat) [inline example/README.md:1]: Example custom top toolbar keybroad

File: example/README.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+This project is a starting point for a Flutter application.
+
+A few resources to get you started if this is your first Flutter project:
+
+- [Lab: Write your first Flutter app](https://docs.flutter.dev/get-started/codelab)
+- [Cookbook: Useful Flutter samples](https://docs.flutter.dev/cookbook)
+
+For help getting started with Flutter development, view the
+[online documentation](https://docs.flutter.dev/), which offers tutorials,
+samples, guidance on mobile development, and a full API reference.
```
Comment (hoangdat) [inline example/README.md:16]: remove

File: example/lib/main.dart [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Flutter Demo Home Page'),
+      ),
+      body: KeyboardRichText(
+        richTextAppendController: richTextAppendController,
+        backgroundKeyboardToolBarColor: Colors.grey,
+        keyBroadToolbar: RichTextKeyboardToolBar(
+            insertImage: () {}, insertAttachment: () {}, appendRickText: () {}),
```
Comment (hoangdat) [inline example/lib/main.dart:71]: how about add Toast for each action?

Comment (hoangdat) [reply example/lib/main.dart:71]: how about an example for customize the toolbar? or let user know in ReadMe

File: example/test/widget_test.dart [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+    // Tap the '+' icon and trigger a frame.
+    await tester.tap(find.byIcon(Icons.add));
+    await tester.pump();
+
+    // Verify that our counter has incremented.
+    expect(find.text('0'), findsNothing);
+    expect(find.text('1'), findsOneWidget);
+  });
+}
```
Comment (hoangdat) [inline example/test/widget_test.dart:30]: remove

File: lib/richtext_append_controller.dart [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import 'dart:async';
+
+class RichTextAppendController {
+  final StreamController<bool> richTextAppendStreamController = StreamController<bool>();
```
Comment (hoangdat) [inline lib/richtext_append_controller.dart:4]: please expose the dispose function of this controller to let user dispose the stream. WDYT?

File: lib/views/widgets/rich_text_keyboard_toolbar.dart [hunk 9]
```diff
@@ -0,0 +1,87 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:rich_text_composer/views/commons/image_paths.dart';
+
+typedef IconWebCallback = void Function();
```
Comment (hoangdat) [inline lib/views/widgets/rich_text_keyboard_toolbar.dart:5]: What is this? As I understand, we only support mobile/tablet. Why related to `web`

File: README.md [hunk 10]
```diff
@@ -1 +1,72 @@
...
+  }
+
+  @override
+  Widget build(BuildContext context) {
+
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Flutter Demo Home Page'),
+      ),
+      body: KeyboardRichText(
```
Comment (hoangdat) [inline README.md:59]: user need to wrap the whole body with our widget? Am I right? How about if user want to add ScrollView?

**Pull request closed without merging**